app/models/ems_refresh.rb

```diff
@@ -116,20 +116,32 @@ def self.get_ar_objects(target, single_id = nil)
     targets_by_type = target.each_with_object(Hash.new { |h, k| h[k] = [] }) do |(c, id), h|
       # Take care of both String or Class type being passed in
       c = c.to_s.constantize unless c.kind_of?(Class)
-      if [VmOrTemplate, Host, ExtManagementSystem].none? { |k| c <= k }
+      if [VmOrTemplate, Host, ExtManagementSystem, ManagerRefresh::Target].none? { |k| c <= k }
         _log.warn "Unknown target type: [#{c}]."
         next
       end
 
       h[c] << id
     end
 
-    # Do lookups to get ActiveRecord objects
+    # Do lookups to get ActiveRecord objects or initialize ManagerRefresh::Target for ids that are Hash
     targets_by_type.each_with_object([]) do |(c, ids), a|
```

Comment:
right, I tried to leave minimal impact to the existing code (so I don't break something :-) )